pypesto/problem.py

```diff
@@ -77,10 +86,13 @@ def __init__(self,
                  lb: Union[np.ndarray, List[float]],
                  ub: Union[np.ndarray, List[float]],
                  dim_full: Optional[int] = None,
-                 x_fixed_indices: Optional[Sequence[int]] = None,
-                 x_fixed_vals: Optional[Sequence[float]] = None,
-                 x_guesses: Optional[Sequence[float]] = None,
-                 x_names: Optional[Sequence[str]] = None):
+                 x_fixed_indices: Optional[Iterable[int]] = None,
```

Comment:
would rather assume those are `Sequence` not `Iterable`. Iterables are single-use, Sequences are multi-use.

Reply:
yes, Sequence. Due to numpy not being of sequence type, that is also not strictly correct as we want to allow numpy input, but I guess that would be fine.